wikibot/wiki.py

```diff
@@ -1,28 +1,59 @@
 from hamper.interfaces import ChatCommandPlugin, Command
-import requests as re
-import json
+import requests
 import os
 
+
 class WikiBot(ChatCommandPlugin):
     name = 'wikibot'
-
-    class TestCommand(Command):
+    api_base_url = 'https://en.wikipedia.org/w/api.php'
+
+    def _api_call(self, **kwargs):
+        """Make an API call to Wikipedia, and returned the data returned."""
+        params = {
+            'format': 'json',
+        }
+        params.update(kwargs)
+        r = requests.get(self.api_base_url, params=params)
+        return r.json()
+
+    class WikiSummaryCommand(Command):
         regex = 'wiki (.+)'
 
         def command(self, bot, comm, groups):
             """
             !wiki <query> -> wiki summary of <query>
+
+            Uses wikipedia api to grab the first 250 or so characters of
+            `query` article (or lets you know the query is ambigious) and
+            appends a clickable url at the end.
             """
             query = groups[0]
-            bot.reply(comm, (self.summary(query, comm)))
 
+            # Generates list of flags and strips out said flags from input
+            query, flag_list = self.flags(query)
+
+            if 'help' in flag_list:
+                return self.print_helptext()
+
+            # Generates psuedo-slugified url
+            url = 'https://en.wikipedia.org/wiki/' + query.replace(' ', '_')
 
-        def summary(self, query, comm):
-            """Returns wikipedia summary of <query>
+            # Get the article summary
+            summary = self.get_article_summary(query)
+            if summary is None:
+                bot.reply(comm, "{user}: I couldn't find an article for {query}",
+                          kwvars={'query': query})
+                return
 
-            Uses wikipedia api to grab the first <280 characters of <query>
-            article (or lets you know the query is ambigious) and appends a
-            clickable url at the end.
+            # Done!
+            bot.reply(comm, '{user}: {summary} :: URL: {url}',
+                      kwvars={'summary': summary, 'url': url})
```

Comment:
Funny, I was just thinking about ripping that feature out. It is hardly used, and because it isn't documented and isn't widely known it seems likely to bite people in the ass.